Mustachio/Tokenizer.cs

```diff
@@ -51,117 +51,135 @@ private static CharacterLocation HumanizeCharacterLocation(string content, int c
             return retval;
         }
 
-        public static IEnumerable<TokenPair> Tokenize(string templateString)
+        public static TokenizeResult Tokenize(string templateString, ParsingOptions parsingOptions)
         {
             templateString = templateString ?? "";
             var matches = _tokenFinder.Matches(templateString);
-            var scopestack = new Stack<Tuple<string, int>>();
+            var scopeStack = new Stack<Tuple<string, int>>();
+            var sourceName = parsingOptions.SourceName;
 
             var idx = 0;
 
+            var tokens = new List<TokenTuple>();
             var parseErrors = new List<IndexedParseException>();
             int[] lines = null;
 
+            var lowPrecedenceExpanders = parsingOptions.TokenExpanders?.Where(e => e.Precedence == Precedence.Low).ToList() ?? new List<TokenExpander>();
```

Comment:
Use `var expanders = parsingOptions.TokenExpanders.ToLookup(k=>k.Precedence)` instead, then you don't need to track these variables as closely later on, just iterate over `expanders[Precedence.Medium]`, etc. when required.